omnia_timeseries_sdk/client.py

```diff
@@ -142,47 +124,114 @@ def _do_request(self, method: str, resource: str, version: str, endpoint: str, p
         # request new access token
         self._token_request()
 
-        url_path = "/" + "/".join([p for p in [resource, version, endpoint] if p.strip()])
+        url = "/" + "/".join([p for p in [resource, version, endpoint] if p.strip()])
         if parameters is not None and isinstance(parameters, dict):
+            parameters = to_camel_case({k: v for k, v in parameters.items() if v is not None})
             enc_parameters = urllib.parse.urlencode(parameters)
+            limit = parameters.get("limit")
         else:
             enc_parameters = ""
+            limit = None
 
-        full_path = f"{url_path}/?{enc_parameters}"
+        url_with_parameters = f"{url}/?{enc_parameters}"
         headers = dict(
             Authorization=f"Bearer {os.getenv('currentOmniaAccessToken', '')}",
             Connection="keep-alive",
             Host=BASE_URL,
         )
-        if body is not None:
-            headers["Content-Type"] = "application/json; charset=utf=8"
 
-        msg = f"{method.upper()} {full_path} {http.client.__doc__.split()[0]}"
+        msg = f"{method.upper()} {url_with_parameters} {http.client.__doc__.split()[0]}"
         for k, v in headers.items():
             msg += f"\n{k}: {v}"
 
         if body is not None:
+            body = to_camel_case({k: v for k, v in body.items() if v is not None})
+            headers["Content-Type"] = "application/json"
             msg += f"\nBody:\n{json.dumps(body, indent=2)}"
 
         logging.debug(msg)
 
         try:
             connection = http.client.HTTPSConnection(self.base_url)
```

Comment:
Is there any reason not to use the `request` package? It is kind of the default choice for these things, to the point that even the documentation for `http.client.HTTPSConnection` sais  
> The Requests package is recommended for a higher-level HTTP client interface. 

With it you [wont have to craft the query_url manually](https://requests.readthedocs.io/en/master/user/quickstart/#passing-parameters-in-urls)

Reply:
Not a very good reason. I started this project with [this](https://api.equinor.com/docs/services/Timeseries-api-v1-5/operations/getTimeseries?) sample Python code and I have not prioritized to replace it with the `request` package. But that has been in the back of my mind all the way because I, as you, favour `request`.

Reply:
I will improve this once when I get time. See #6